cmd/flow/main.go

```diff
@@ -20,46 +20,126 @@
+	"github.com/onflow/flow-cli/internal/cadence"
+	"github.com/onflow/flow-cli/internal/collections"
+	"github.com/onflow/flow-cli/internal/events"
+	"github.com/onflow/flow-cli/internal/keys"
+	"github.com/onflow/flow-cli/internal/project"
+	"github.com/onflow/flow-cli/internal/scripts"
+	"github.com/onflow/flow-cli/internal/transactions"
+	"github.com/onflow/flow-cli/internal/version"
 )
 
-var cmd = &cobra.Command{
+var c = &cobra.Command{
 	Use:              "flow",
 	TraverseChildren: true,
 }
 
 func init() {
-	cmd.AddCommand(project.Cmd)
-	cmd.AddCommand(initialize.Cmd)
-	cmd.AddCommand(accounts.Cmd)
-	cmd.AddCommand(blocks.Cmd)
-	cmd.AddCommand(collections.Cmd)
-	cmd.AddCommand(keys.Cmd)
-	cmd.AddCommand(emulator.Cmd)
-	cmd.AddCommand(events.Cmd)
-	cmd.AddCommand(cadence.Cmd)
-	cmd.AddCommand(scripts.Cmd)
-	cmd.AddCommand(transactions.Cmd)
-	cmd.AddCommand(version.Cmd)
-	cmd.PersistentFlags().StringSliceVarP(&cli.ConfigPath, "config-path", "f", cli.ConfigPath, "Path to flow configuration file")
+
+	c.AddCommand(cadence.Cmd)
+	c.AddCommand(version.Cmd)
+	c.AddCommand(emulator.Cmd)
+
+	c.AddCommand(accounts.Cmd)
+	command.Add(accounts.Cmd, accounts.NewGetCmd())
+	command.Add(accounts.Cmd, accounts.NewCreateCmd())
+	command.Add(accounts.Cmd, accounts.NewAddContractCmd())
+	command.Add(accounts.Cmd, accounts.NewRemoveContractCmd())
+	command.Add(accounts.Cmd, accounts.NewUpdateContractCmd())
+
+	c.AddCommand(scripts.Cmd)
+	command.Add(scripts.Cmd, scripts.NewExecuteScriptCmd())
+
+	c.AddCommand(transactions.Cmd)
+	command.Add(transactions.Cmd, transactions.NewSendCmd())
+	command.Add(transactions.Cmd, transactions.NewStatusCmd())
+
+	c.AddCommand(keys.Cmd)
+	command.Add(keys.Cmd, keys.NewGenerateCmd())
+	command.Add(keys.Cmd, keys.NewCmdDecode())
+
+	c.AddCommand(events.Cmd)
+	command.Add(events.Cmd, events.NewGetCmd())
+
+	c.AddCommand(blocks.Cmd)
+	command.Add(blocks.Cmd, blocks.NewGetCmd())
+
+	c.AddCommand(collections.Cmd)
+	command.Add(collections.Cmd, collections.NewGetCmd())
+
+	c.AddCommand(project.Cmd)
+	command.Add(project.Cmd, project.NewInitCmd())
+	command.Add(project.Cmd, project.NewDeployCmd())
+
+	c.PersistentFlags().StringVarP(
+		&command.HostFlag,
+		"host",
+		"",
+		command.HostFlag,
+		"Flow Access API host address",
+	)
+
+	c.PersistentFlags().StringVarP(
+		&command.FilterFlag,
+		"filter",
+		"",
+		command.FilterFlag,
+		"Filter result values by property name",
+	)
+
+	c.PersistentFlags().StringVarP(
+		&command.FormatFlag,
+		"format",
+		"",
+		command.FormatFlag,
+		"Format to show result in",
+	)
+
+	c.PersistentFlags().StringVarP(
+		&command.SaveFlag,
+		"save",
+		"",
+		command.SaveFlag,
+		"Save result to a filename",
+	)
+
+	c.PersistentFlags().StringVarP(
+		&command.LogFlag,
+		"log",
+		"",
+		command.LogFlag,
+		"Logging level",
+	)
+
+	c.PersistentFlags().BoolVarP(
+		&command.RunEmulatorFlag,
+		"emulator",
+		"",
+		command.RunEmulatorFlag,
+		"Run in-memory emulator",
+	)
```

Comment:
This flag only applies to commands that would normally use the Access API, right?

Reply:
Yes. I think with your previous point we could have a subgroup of commands that have some flags (save, emulator etc). I was anyhow thinking this global command should have a structure on its own probably so we don't have to add all the commands in main but they would just be added to that structure, but I didn't have the time to solve this, maybe we can together discuss this, however not necessary to do it now.